report/search_incidents.go

```diff
@@ -1,397 +1,39 @@
-
+func searchIncidents(ctx context.Context, client *datadog.APIClient, r *searchRequest) (datadogV2.IncidentSearchResponse, *nethttp.Response, error) {
 	var queryOpts []string
-
 	if r.createdBefore != nil {
 		queryOpts = append(queryOpts, fmt.Sprintf("created_before:%d", *r.createdBefore))
 	}
 	if r.createdAfter != nil {
 		queryOpts = append(queryOpts, fmt.Sprintf("created_after:%d", *r.createdAfter))
 	}
 	queryOpts = append(queryOpts, r.tags...)
+	query := strings.Join(queryOpts, " ")
 
-	if len(queryOpts) > 0 {
-		localVarQueryParams.Add("query", strings.Join(queryOpts, " "))
-	}
-
-	if r.pageSize != nil {
-		localVarQueryParams.Add("page[size]", parameterToString(*r.pageSize, ""))
-	}
-	if r.pageOffset != nil {
-		localVarQueryParams.Add("page[offset]", parameterToString(*r.pageOffset, ""))
-	}
-
-	localVarQueryParams.Add("filter[field_type]", "all")
-	localVarQueryParams.Add("sort", "created")
-
-	// to determine the Content-Type header
-	localVarHTTPContentTypes := []string{}
-
-	// set Content-Type header
-	localVarHTTPContentType := selectHeaderContentType(localVarHTTPContentTypes)
-	if localVarHTTPContentType != "" {
-		localVarHeaderParams["Content-Type"] = localVarHTTPContentType
-	}
-
-	// to determine the Accept header
-	localVarHTTPHeaderAccepts := []string{"application/json"}
-
-	// set Accept header
-	localVarHTTPHeaderAccept := selectHeaderAccept(localVarHTTPHeaderAccepts)
-	if localVarHTTPHeaderAccept != "" {
-		localVarHeaderParams["Accept"] = localVarHTTPHeaderAccept
-	}
-
-	// Set Operation-ID header for telemetry
-	localVarHeaderParams["DD-OPERATION-ID"] = "SearchIncidents"
-
-	if ctx != nil {
-		// API Key Authentication
-		if auth, ok := ctx.Value(datadog.ContextAPIKeys).(map[string]datadog.APIKey); ok {
-			if apiKey, ok := auth["apiKeyAuth"]; ok {
-				var key string
-				if apiKey.Prefix != "" {
-					key = apiKey.Prefix + " " + apiKey.Key
-				} else {
-					key = apiKey.Key
-				}
-				localVarHeaderParams["DD-API-KEY"] = key
-			}
-		}
-	}
-	if ctx != nil {
-		// API Key Authentication
-		if auth, ok := ctx.Value(datadog.ContextAPIKeys).(map[string]datadog.APIKey); ok {
-			if apiKey, ok := auth["appKeyAuth"]; ok {
-				var key string
-				if apiKey.Prefix != "" {
-					key = apiKey.Prefix + " " + apiKey.Key
-				} else {
-					key = apiKey.Key
-				}
-				localVarHeaderParams["DD-APPLICATION-KEY"] = key
-			}
-		}
-	}
-	req, err := client.PrepareRequest(ctx, localVarPath, localVarHTTPMethod, localVarPostBody, localVarHeaderParams, localVarQueryParams, localVarFormParams, localVarFormFileName, localVarFileName, localVarFileBytes)
-	if err != nil {
-		return localVarReturnValue, nil, err
-	}
-
-	localVarHTTPResponse, err := client.CallAPI(req)
-	if err != nil || localVarHTTPResponse == nil {
-		return localVarReturnValue, localVarHTTPResponse, err
-	}
+	queryParams := datadogV2.NewSearchIncidentsOptionalParameters().WithSort(datadogV2.INCIDENTSEARCHSORTORDER_CREATED_ASCENDING)
```

Comment:
This matches the sort order specified by the previous query: https://github.com/xornivore/incidentist/blob/33efc1adfbc0599a810e5e618c26307ee03449ba/report/search_incidents.go#L270